cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,821 @@
+      # ${${container}elf_files} as dependency to ensure they are loaded by SES
+      # even though it is unnecessary for building the application.
+      ${${container}elf_files}
+      )
+
+    # Wrapper target for the merge command.
+    add_custom_target(
+      ${container}_hex
+      ALL DEPENDS
+      ${PROJECT_BINARY_DIR}/${container}.hex
+      )
+
+    if (DEFINED PM_DOMAIN)
+      update_runner(IMAGE ${DOMAIN_APP_${PM_DOMAIN}} HEX ${PROJECT_BINARY_DIR}/${container}.hex)
+    endif()
+
+    if ("${container}" STREQUAL "merged")
+      update_runner(IMAGE ${DEFAULT_IMAGE} HEX ${PROJECT_BINARY_DIR}/${container}.hex)
+    endif()
+  endforeach()
+
+endfunction()
+
+# Function to update an already generated runners file for a Zephyr image.
+# This function allows the build system to adjust files to flash in the config
+# section of the runners.yaml on a given image.
+function(update_runner)
+  cmake_parse_arguments(RUNNER "" "IMAGE;HEX;BIN;ELF" "" ${ARGN})
+
+  check_arguments_required("update_runner" RUNNER IMAGE)
+
+  get_target_property(bin_dir ${RUNNER_IMAGE} _EP_BINARY_DIR)
+  set(runners_file ${bin_dir}/zephyr/runners.yaml)
+
+  set(runners_content_update)
+  file(STRINGS ${runners_file} runners_content)
+  foreach(line ${runners_content})
+    if(DEFINED RUNNER_ELF AND ${line} MATCHES "^.*elf_file: .*$")
+      string(REGEX REPLACE "(.*elf_file:) .*" "\\1 ${RUNNER_ELF}" line ${line})
+    endif()
+
+    if(DEFINED RUNNER_HEX AND ${line} MATCHES "^.*hex_file: .*$")
+      string(REGEX REPLACE "(.*hex_file:) .*" "\\1 ${RUNNER_HEX}" line ${line})
+    endif()
+
+    if(DEFINED RUNNER_BIN AND ${line} MATCHES "^.*bin_file: .*$")
+      string(REGEX REPLACE "(.*bin_file:) .*" "\\1 ${RUNNER_BIN}" line ${line})
+    endif()
+    list(APPEND runners_content_update "${line}\n")
+  endforeach()
+  file(WRITE ${runners_file} ${runners_content_update})
+endfunction()
+
+
+# APP is a special domain which is handled differently.
+# Remove it from the list.
+list(REMOVE_ITEM PM_DOMAINS APP)
+
+set(user_def_pm_static ${PM_STATIC_YML_FILE})
+
+ncs_file(CONF_FILES ${APPLICATION_CONFIG_DIR}
+         PM conf_dir_pm_static
+         DOMAIN ${DOMAIN}
+         BUILD ${CONF_FILE_BUILD_TYPE}
+)
+
+ncs_file(CONF_FILES ${BOARD_DIR}
+         PM board_dir_pm_static
+         DOMAIN ${DOMAIN}
+         BUILD ${CONF_FILE_BUILD_TYPE}
+)
+
+if(EXISTS "${user_def_pm_static}" AND NOT IS_DIRECTORY "${user_def_pm_static}")
+  set(static_configuration_file ${user_def_pm_static})
+elseif (EXISTS ${conf_dir_pm_static})
+  set(static_configuration_file ${conf_dir_pm_static})
+elseif (EXISTS ${board_dir_pm_static})
+  set(static_configuration_file ${board_dir_pm_static})
+endif()
+
+if (EXISTS ${static_configuration_file})
+  message(STATUS "Found partition manager static configuration: "
+                 "${static_configuration_file}"
+  )
+  set(static_configuration --static-config ${static_configuration_file})
+endif()
```

Comment:
This part should probably be moved, because each domain is supposed to use a different static configuration file. Using the same file for all domains can produce errors, as with:
```
west build -b nrf5340dk_nrf5340_cpuapp samples/matter/lock -p --sysbuild \
-DSB_CONFIG_NETCORE_HCI_RPMSG=y \
-DSB_CONFIG_BOOTLOADER_MCUBOOT=y \
-DSB_CONFIG_SECURE_BOOT=y \
-DPM_STATIC_YML_FILE=samples/matter/lock/configuration/nrf5340dk_nrf5340_cpuapp/pm_static_dfu.yml

...

Partition manager failed: Incorrect amount of gaps found in static configuration. (...)
```
The provided `PM_STATIC_YML_FILE` is only meant for the main app, but now sysbuild applies it for `hci_rpmsg` here too, which is an error.

Note that namespacing `PM_STATIC_YML_FILE` should probably also be supported:

https://github.com/nrfconnect/sdk-nrf/blob/24ea7f01cc8fe1fd9dcbabddb2b82c98d0fc8832/applications/nrf5340_audio/CMakeLists.txt#L28-L29

Reply:
fixed.

I kept the `<domain-app>_PM_STATIC_YML_FILE` usage, as in your example `mcuboot_PM_STATIC_YML_FILE`, but with the introduction of sysbuild I wonder if we should use `<domain>_PM_STATIC_YML_FILE` instead, like `CPUNET_PM_STATIC_YML_FILE`.

The original reason for the `<domain-app>_PM_STATIC_YML_FILE` was due to pm running in the parent image scope of the domain app and thus the `<domain_app>` prefix would ensure the CMake variable would be passed to the related image build, but with sysbuild then all partition manager handling is done in sysbuild scope and therefore we can use the domain name as prefix instead of the app name.
That will also indicate that the `CPUNET` is not being passed as variable to the image build.

I will not do such change in this PR, but I think it's a valid discussion to start.